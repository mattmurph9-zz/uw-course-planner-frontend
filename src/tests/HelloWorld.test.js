import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import HelloWorld from '../components/HelloWorld';

describe(HelloWorld, () => {
    const name = 'Person';
    const mockRemoveGreeting = jest.fn();
    const component = shallow(
        <HelloWorld name={name} removeGreeting={mockRemoveGreeting} />
    );
    it('renders and matches snapshot', () => {
        const component = renderer.create(
            <HelloWorld name="Person" />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('contains the supplied name', () => {
        expect(component.text()).toContain(name);
    });
    it('modifies greeting when Change Language button is clicked', () => {
        component.find('button.changeLang').simulate('click');
        expect(component.text()).toContain('Bonjour');
        component.find('button.changeLang').simulate('click');
        expect(component.text()).toContain('Hello');
    });
    it('calls the passed in removeGreeting function when remove button is clicked', () => {
        component.find('button.remove').simulate('click');
        expect(mockRemoveGreeting).toHaveBeenCalled();
    });
});