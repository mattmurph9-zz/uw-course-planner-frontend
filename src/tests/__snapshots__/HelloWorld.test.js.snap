// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class HelloWorld extends _react.Component {
    constructor(props) {
        super(props);this.





        changeLang = () => {
            let greeting;
            if (this.state.greeting === 'Hello') greeting = 'Bonjour';else
            greeting = 'Hello';
            this.setState({ greeting });
        };this.state = { greeting: 'Hello' };}

    render() {
        return (
            _react2.default.createElement('div', { className: 'HelloWorld', __source: { fileName: _jsxFileName, lineNumber: 21 }, __self: this },
                this.state.greeting, ' ', this.props.name,
                _react2.default.createElement('br', { __source: { fileName: _jsxFileName, lineNumber: 23 }, __self: this }),
                _react2.default.createElement('button', { className: 'changeLang', onClick: this.changeLang, __source: { fileName: _jsxFileName, lineNumber: 24 }, __self: this }, 'Change Language'),
                _react2.default.createElement('br', { __source: { fileName: _jsxFileName, lineNumber: 25 }, __self: this }),
                _react2.default.createElement('button', { className: 'remove', onClick: () => this.props.removeGreeting(this.props.name), __source: { fileName: _jsxFileName, lineNumber: 26 }, __self: this }, 'Remove Entry')));




    }} renders and matches snapshot 1`] = `
<div
  className="HelloWorld"
>
  Hello
   
  Person
  <br />
  <button
    className="changeLang"
    onClick={[Function]}
  >
    Change Language
  </button>
  <br />
  <button
    className="remove"
    onClick={[Function]}
  >
    Remove Entry
  </button>
</div>
`;
